doc/specs/#1595 - Suggestions UI/Suggestions-UI.md

```diff
@@ -0,0 +1,734 @@
+> mentioning. This might be better served in the [shell integration] doc._
+
+We'll probably want a way for recent commands to be saved across sessions. That way, your `cmd.exe` command history could persist across sessions. We'd need:
+
+* A setting to enable this behavior
+* A setting to control the context of these saved commandlines.
+  * Do we want them saved per-profile, or globally?
+  * If they're saved per-profile, maybe a profile can opt-in to loading all the commands?
+  * How does defterm play with this? Do we "layer: by concatenating per-profile commands with `profiles.defaults` ones?
+* A button in the Settings UI for clearing these commands
+* Should fragments be able to pre-populate "recent commands"?
+  * I'm just gonna say _no_. That would be a better idea for Tasks (aka just a `sendInput` Action that we load from the fragment normally as a Task), or a specific suggestion source for the fragment extension.
+
+#### Inline mode 
+
+> **Note**
+> _This is a half-baked idea with some potential. However, I don't
+> think it needs to be a part of the v1 of the Suggestions UI, so I'm leaving it
+> under future considerations for a future revision._
+
+Do we want to have a suggestions UI "mode", that's just **one** inline
+suggestion, "no" UI? Some UX ala the `PsReadline` recent command suggestion
+feature. Imagine, we just display the IME ghost text thing for the first result,
+given the current prompt? 
+
+Take the following action as an example:
+
+```json
+{ "command": { "action":"suggestions", "source": "commandHistory", "useCommandline": true, "inline": true } },
+```
+
+Type the start of some command at the prompt, and press that key. Presto, we do
+the `pwsh` thing. Ghost text appears for the first match in the `commandHistory`
+for what the user has typed. If they press another key, ~they've typed into the
+"hidden" Suggestions UI, which filters the (hidden) list more, and updates the
+one inline suggestion.~
+
+Or, instead, typed keys go to the shell, and then we re-query the commandline,
+and update the filter accordingly. That would allow tab-completion to still
+work. We'd use <kbd>right arrow</kbd> to accept the suggestion (and dismiss the
+ghost text preview).
+
+This would seemingly SUPER conflict with PowerShell's own handler. Probably not
+something someone should enable for PowerShell 7 profiles if they're using that
+feature. 
+
+### Rejected ideas
+
+These are musings from earlier versions of the spec.
+* **Asynchronous prompting**: This was rejected because it was so fundamentally
+  different from the rest of the UX of the Suggestions UI, it didn't make sense
+  to try and also do that behavior. 
+  
+  Furthermore, async sources wouldn't work with sync
+  ones, at all. E.g. if you did `source: ["tasks", "myAsyncSource"]`. It doesn't
+  make sense to start with a list of `tasks`, then type, find no tasks, but then
+  oh! the UI fills in some other suggestions too. That's weird. 

```

Comment:
Psychic debugging: you wrote this bullet point, and then later wrote a whole section on it! You may be able to streamline the doc by removing this in favor of the section below.